opcua.cpp

```diff
@@ -1376,6 +1409,66 @@ int OPCUA::subscribe()
 	return (int)status;
 }
 
+/**
+ * Asynchronous service response callback.
+ *
+ * Handles responses from asynchronous S2OPC services, specifically the WriteResponse in this case.
+ *
+ * @param encType     The type of the encodeable message received.
+ * @param response    A pointer to the response data.
+ * @param appContext  The application context passed to the asynchronous request.
+ */
+void OPCUA::asyncS2ResponseCallBack(SOPC_EncodeableType *encType, const void *response, uintptr_t appContext)
+{
+	// Check if the response is a WriteResponse
+	if (encType == &OpcUa_WriteResponse_EncodeableType)
+	{
+		// Cast the response to the appropriate type
+		const OpcUa_WriteResponse *writeResponse = (const OpcUa_WriteResponse *)response;
+
+		// Check the overall service result status
+		if (SOPC_IsGoodStatus(writeResponse->ResponseHeader.ServiceResult))
+		{
+			// Ensure there is exactly one result and check its status
+			if (writeResponse->NoOfResults == 1)
+			{
+				if (SOPC_IsGoodStatus(writeResponse->Results[0]))
+				{
+					Logger::getLogger()->info("Write operation succeeded.");
+				}
+				else
+				{
+					Logger::getLogger()->error(
+						"Write operation failed with result status: 0x%08" PRIX32, writeResponse->Results[0]);
```

Comment:
Again an indication of which write request this is. It is probably not an operation either, the word "operation" has a specific meaning in FogLAMP control. Also can we not print a message rather than a status code that is meaningless to the reader of this error message.